src/io.jl

```diff
@@ -54,24 +54,31 @@ Print to `io` the best 8-bit SGR color code that sets the `category` color to
 be close to `color`.
 """
 function termcolor8bit(io::IO, (; r, g, b)::RGBTuple, category::Char)
-    # Magic numbers? Lots.
-    cdistsq(r1, g1, b1) = (r1 - r)^2 + (g1 - g)^2 + (b1 - b)^2
-    to6cube(value) = if value < 48; 1
-    elseif value < 114; 2
-    else 1 + (value - 35) ÷ 40 end
+    # RGB values are mapped to a 6x6x6 "colour cube", which (mapped to
+    # 24-bit colour space), jumps up from a black level of 0 in each
+    # component to 95, then takes 4 steps of 40 to reach 255.
+    function cdistsq(r2, g2, b2) # The squared "redmean" colour distance function
+        rr = (r + r2) / 2
+        (2 + r/256) * (r - r2)^2 + 4 * (g - g2)^2 + (2 + (255 - rr)/256) * (b - b2)^2
+    end
+    to6cube(value) = (value - 35) ÷ 40
+    from6cube(r6, g6, b6) = 16 + 6^2 * r6 + 6^1 * g6 + 6^0 * b6
+    sixcube = (0, 95:40:255...)
     r6cube, g6cube, b6cube = to6cube(r), to6cube(g), to6cube(b)
-    sixcube = (0, 95, 135, 175, 215, 255)
-    rnear, gnear, bnear = sixcube[r6cube], sixcube[g6cube], sixcube[b6cube]
+    rnear, gnear, bnear = sixcube[r6cube+1], sixcube[g6cube+1], sixcube[b6cube+1]
     colorcode = if r == rnear && g == gnear && b == bnear
-        16 + 35 * r6cube + 6 * g6cube + b6cube
+        from6cube(r6cube, g6cube, b6cube)
```

Comment:
Bless you for these comments, this stuff is wild

Reply:
It feels a lot better to have cleaner code to look at too :sweat_smile: